s3-media-sync.php

```diff
@@ -26,7 +26,13 @@
 	function() {
 		$settings_handler = new S3_Media_Sync_Settings();
 		$settings = $settings_handler->get_settings();
-		$tester = new S3_Media_Sync_Tester($settings);
+		
+		// Only create the tester if we have the required settings configured
+		// This prevents errors when the plugin is first activated with no settings
+		if ( $settings_handler->has_required_settings() ) {
+			$tester = new S3_Media_Sync_Tester($settings);
```

Comment:
Consider keeping registration work (CLI commands, admin menus, hooks) outside this conditional so the UX is intact even when config is incomplete. Only gate code paths that require an initialized bucket or network access.